examples/sherpa3d.py

```diff
@@ -0,0 +1,170 @@
+    """Combine a spatial and spectral model into a 3D model.
+    """
+
+    def __init__(self, spatial_model, spectral_model, name='model'):
+        # Store references to the parts (not really needed?)
+        self.spatial_model = spatial_model
+        self.spectral_model = spectral_model
+
+        # Concatenate parameters
+        pars = spatial_model.pars + spectral_model.pars
+
+        ArithmeticModel.__init__(self, name, pars)
+
+    def calc(self, pars, *args, **kwargs):
+        # c_x, c_y, c_e = pars
+        # import IPython; IPython.embed()
+        print('pars: ', pars)
+        print('args: ', args)
+        print('kwargs: ', kwargs)
+
+        # TODO: really we should compute the integral flux over the energy bin,
+        # and then set the norm of the spatial model so that it corresponds to that flux.
+
+        # spectral_term = self.spatial_model(x, y)
+        # spatial_term = self.spectral_model(energy)
+        # value = spectral_term * spatial_term
+        value = 290
+
+        return value
+
+
+def test_FromScratchModel3D():
+    print('Running test_FromScratchModel3D')
+    model = FromScratchModel3D()
+    print(model)
+    actual = model(20, 30, 40)
+    expected = (2 * 20) + (3 * 30) + (4 * 40)
+    assert_allclose(actual, expected)
+
+
+def test_CombinedModel3D():
+    print('Running test_CombinedModel3D')
+    spatial_model = Polynom2D()
+    spatial_model.c = 0
+    spatial_model.cx1 = 2
+    spatial_model.cy1 = 3
+    spectral_model = Polynom1D()
+    spectral_model.c0 = 0
+    spectral_model.c1 = 4
+
+    model = CombinedModel3D(spatial_model, spectral_model)
+    print(model)
+    actual = model(20, 30, 40)
+    expected = (2 * 20) + (3 * 30) + (4 * 40)
+    assert_allclose(actual, expected)
+
+
+def run_cash_fit(data, model):
+    fit = Fit(data=data, model=model, stat=Cash(), method=LevMar(), estmethod=Covariance())
+    result = fit.fit()
+    result_err = fit.est_errors()
+
+    print('=> data:')
+    print(data)
+    print('=> model:')
+    print(model)
+    print('=> fit:')
+    print(fit)
+    print('=> result:')
+    print(result)
+    print('=> result_err:')
+    print(result_err)
+
+
+def test_fit_1D():
+    x = np.array([1, 2, 10])
+    y = np.array([10, 30, 20])
+    data = Data1D(name='data', x=x, y=y)
+    model = Const1D()
+    run_cash_fit(data, model)
+
+
+def test_fit_2D():
+    x0 = np.array([[1, 2, 10], [1, 2, 10]])
+    x1 = np.array([[1, 2, 10], [1, 2, 10]])
+    y = np.array([[10, 30, 20], [10, 30, 20]])
+    data = Data2D(name='data', x0=x0.flatten(), x1=x1.flatten(), y=y.flatten())
+    model = Const2D()
+    run_cash_fit(data, model)
+
+
+def test_fit_1D_ND():
+    """Same as with Data1D, but using DataND class.
+    """
+    x = np.array([1, 2, 10])
+    y = np.array([10, 30, 20])
+    # x0=x0.flatten(), x1=x1.flatten(), y=y.flatten()
+    indep = (x, )
+    dep = (y, )
+    import IPython; IPython.embed()
+    data = DataND(name='data', indep=indep, dep=dep)
```

Comment:
@anetasie This is the first of probably many questions I have for you how to implement this. Can / should I use `DataND` directly or should I sub-class `DataND` to a `Data3D` class?
Using `DataND` directly doesn't seem to work, I get an [error](https://gist.github.com/cdeil/6db3e885bff400933fea#file-gistfile1-txt-L49):
```
AttributeError: 'DataND' object has no attribute 'y'
```